deals/types.go

```diff
@@ -20,8 +18,8 @@ type StoreResult struct {
 	Message     string
 }
 
-// DealInfo contains information about a proposed storage deal.
-type DealInfo struct {
+// StorageDealInfo contains information about a proposed storage deal.
+type StorageDealInfo struct {
```

Comment:
Making some names in here more succinct now that retrieval deal info and records are a thing